### DISCUSSION TITLE
Venus-OS Merged Battery

### DISCUSSION THREAD

**Discussion by dominikandreas** [mr-manuel/venus-os_dbus-mqtt-pv#44] [Show and tell]
Hey @mr-manuel , just wanted to say thanks for your work. I've taken some parts from this repository to build my "Venus-OS-Merged-Battery" driver which merges information from a BMS and a SmartShunt for a single virtual battery that benefits from the detailed information from the BMS and the precise measurements from the SmartShunt. In case you're interested, you can find it here: https://github.com/dominikandreas/venus-os_merged-battery

Cheers,
Dominik

**Comment by mr-manuel** [comment]
From this repository? Are you sure? This has nothing to do with battery. Maybe you mean [dbus-mqtt-battery](https://github.com/mr-manuel/venus-os_dbus-mqtt-battery)? Anyway that what you plan to do is already possible with [dbus-serialbattery](https://github.com/mr-manuel/venus-os_dbus-serialbattery).

**Comment by dominikandreas** [comment]
Not the technical implementation, but the installation scripts, etc ;-) 
Are you sure? I thought dbus-serialbattery is a driver for connecting unsupported BMS via serial, in my case the data is already on the dbus, the merges the data from two batteries from dbus and republishes them.

**Reply by mr-manuel** [reply]
The installation scripts are all the same in my dbus-mqtt-* repos :-)

If you already use dbus-serialbattery you can use: https://github.com/mr-manuel/venus-os_dbus-serialbattery/blob/a7d7868555b37a4ba5c1f7ffd1df518fbbe72c0a/dbus-serialbattery/config.default.ini#L205-L216

**Reply by dominikandreas** [reply]
That's good to know, thanks for the pointer.

I don't use dbus-serialbattery in my setup since the new JK-BMS supports Victron natively via CAN, no driver needed.

**Reply by mr-manuel** [reply]
Natively is not correct, since you use probably the Pylontech protocol to get it. Check the nightly build of the dbus-serialbattery. It gets the data from the JKBMS CAN protocol and fetches all the data including single cell voltages and not only min/max. In addition you get a better and smoother battery handling, which increases battery life.

**Reply by dominikandreas** [reply]
Well natively in the sense that I simply select "victron protocol" in the BMS setting and plug in the cable. But you're right, I do get only min-max voltages, number of packs online / blocking charge / discharge and CVL/CCL. I'm not worried too much about the smoother handling, with 2 x 314Ah packs and the current ESS assistant config I'm practically far from unsafe currents anyway. Having the individual cell voltages in victron would be nice though, at the moment I only have those in home assistant via esp32-ble and esphome.
